crates/compiler/solve/src/solve.rs

```diff
@@ -2243,8 +2230,13 @@ fn either_type_index_to_var(
                 pools,
                 types,
                 aliases,
-                typ_cell,
-            )
+                type_index,
+            );
+            unsafe {
+                types.emplace_variable(type_index, var);
+            }
```

Comment:
maybe add a comment why the conditions on `emplace_variable` are met here

Reply:
I'll send you a follow up after this to avoid kicking CI on this PR again